File: server/src/com/inkostilation/pong/server/network/IProcessor.java
package com.inkostilation.pong.server.network;

import com.inkostilation.pong.commands.AbstractResponseCommand;

import java.nio.channels.SocketChannel;

public interface IProcessor {

    void start();
    void stop();
    void processConnections();
    void sendMessage(AbstractResponseCommand command, SocketChannel channel);
}

File: core/src/com/inkostilation/pong/server/network/IProcessor.kt
package com.inkostilation.pong.server.network

import com.inkostilation.pong.processing.IStateFul
import java.util.*

interface IProcessor: IStateFul {
    fun start(router: AbstractCommandRouter)
    fun processConnection()
    fun closeConnection(marker: UUID)
    fun stop()
}